### DISCUSSION TITLE
Relaunch of the Ansible role "paperless-ngx/ansible"

### DISCUSSION THREAD

**Discussion by stevenengland** [paperless-ngx/ansible#26] [Announcements]
As a consequence of https://github.com/paperless-ngx/ansible/issues/24 this role will be relaunched with a quite different source base.

I will pull in code from https://github.com/stevenengland/paperless_ngx_ansible_role what means breaking changes to users of the former source base. I hope in the end the overall benefit of this breaking change is greater than the annoyance :-) In return the role will be listed at Ansible Galaxy and will support installations of the current release (1.13.0) as well as it will support all the configuration variables available today. Feel free to have a look at the old repo mentioned above to get an impression.

At the moment I am preparing a few steps as a preparation. 

Stay tuned ;)

**Comment by stevenengland** [comment]
Done by #27 

In the meanwhile a first release was published:
https://github.com/paperless-ngx/ansible/releases/tag/1.0.1
https://galaxy.ansible.com/paperless_ngx/paperless_ngx

**Comment by Tronde** [comment]
Hello @stevenengland,  
Congratulations on the successful relaunch and the two releases of this Ansible role.

I hope you don't mind answering my following questions regarding the current state of this role and where it's heading.

- The role automates the installation steps of the [Bare Metal Route](https://docs.paperless-ngx.com/setup/#bare_metal), right?
- Does it focus on Day-1-Tasks (i.e. Initial setup and configuration) only, or could it be used for Day-2-Tasks (i.e. updates, backup, restore, etc.) as well?
- Currently Debian and Ubuntu are supported. Do you have any plans to support additional distributions, e.g. CentOS Stream, RHEL, SLES etc. in some future release?

Looking forward to reading your answer.

Best regards,  
Tronde

**Reply by stevenengland** [reply]
Hi @Tronde thank you for the warm words ;) And don't mind ask as many questions as you like :-D Here are my answers:

* The role automates the bare metal route, correct
* The role is not limited to day 1 tasks. It supports fresh installations as well as upgrades and plain reconfigurations. On the other hand it does not support backup/restore at all -> on purpuse. See a recent issue with discussion regarding this topic here: #1 I will add a paragraph in the docs to clarify all these points. Thank you for bringing this to my attention..
*  My plans are highly connected to user demands :) As I am personally only using Debian and the original role also focused on Debian I did not expand the scope yet. If there is a demand for that I will have a look into this. Let me know.

**Reply by Tronde** [reply]
Hi @stevenengland,  
Thanks for your quick answer.

>     * The role is not limited to day 1 tasks. It supports fresh installations as well as upgrades and plain reconfigurations. On the other hand it does not support backup/restore at all -> on purpuse. See a recent issue with discussion regarding this topic here: [[PR] extend the backup mechanism during updates #1](https://github.com/paperless-ngx/ansible/issues/1) I will add a paragraph in the docs to clarify all these points. Thank you for bringing this to my attention.

With upgrades do you mean deploying new paperless-ngx releases, right? How often have you successfully done that, yet?  
I assume it saves a lot of time running the role again than following an upgrade guide step by step.

Do you have any experiences whether it could cause any trouble running `apt update && apt upgrade` or `apt update && apt dist-upgrade` on the host where paperless-ngx is installed? Has that lead to any issues in the past since you are using paperless-ngx?

>     * My plans are highly connected to user demands :) As I am personally only using Debian and the original role also focused on Debian I did not expand the scope yet. If there is a demand for that I will have a look into this. Let me know.

I'm using mainly Fedora and RHEL these days, but it's no problem to deploy a Debian VM to host paperless-ngx. So no need to support additional Distributions just for me. ;-)  
Better to concentrate on what you use by yourself and do that good than be a jack of all trades but a master of none. :-)

**Reply by stevenengland** [reply]
Hi @Tronde .

> With upgrades do you mean deploying new paperless-ngx releases, right?

That's correct. To seperate concerns I normally don't touch any other aspect of the underlying system (or let's call it cross sectional tasks) with Ansible roles. 

> How often have you successfully done that, yet?

I upgraded from 1.11.3 to 1.12.0/1/2 and then to 1.13.0 without issues.

> I assume it saves a lot of time running the role again than following an upgrade guide step by step.

I guess so. Although an upgrade in the narrower sense is nothing big: Switch the release files alltogether and execut the database migration command. If you call the role with `latest` for the version var then it is just a matter of calling the role again and again for every new release ;) 

> Do you have any experiences whether it could cause any trouble running apt update && apt upgrade or apt update && apt dist-upgrade on the host where paperless-ngx is installed? Has that lead to any issues in the past since you are using paperless-ngx?

Until now I did not recognize any issues. For that task I use the `unattended-upgrades` BUT I do exclude some of the dependent packages (postgresql and I think some of the deps mentioned in the bare metal route guide -> can look them up later if you want). Purely precautionary ... Those I do upgrade "attended" with extra backups before although I did not had any issues so far and might add them to the "unattended" upgrades...

> Better to concentrate on what you use by yourself and do that good than be a jack of all trades but a master of none. :-)

Good point ;)

**Reply by Tronde** [reply]
Hi @stevenengland,  
I have a follwo-up question. :-)

> > Do you have any experiences whether it could cause any trouble running apt update && apt upgrade or apt update && apt dist-upgrade on the host where paperless-ngx is installed? Has that lead to any issues in the past since you are using paperless-ngx?
> 
> Until now I did not recognize any issues. For that task I use the `unattended-upgrades` BUT I do exclude some of the dependent packages (postgresql and I think some of the deps mentioned in the bare metal route guide -> can look them up later if you want). Purely precautionary ... Those I do upgrade "attended" with extra backups before although I did not had any issues so far and might add them to the "unattended" upgrades...

Could you list the packages you've excluded from from the unattended-upgrades? That would be nice and apprechiated. :-)

Best regards,  
Tronde
